plugin/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/config/ConfigureApplicationAction.java

```diff
@@ -50,10 +52,16 @@ public void actionPerformed(ActionEvent e) {
         presenter.show();
     }
 
+
     @Override
     public void updateInPerspective(@NotNull ActionEvent event) {
-        final CurrentProject currentProject = appContext.getCurrentProject();
-        event.getPresentation().setVisible(true);
-        event.getPresentation().setEnabled(currentProject != null && currentProject.getRootProject().getMixins().contains(OPENSHIFT_PROJECT_TYPE_ID));
+        final Resource resource = appContext.getResource();
+        if (resource != null) {
+            final Optional<Project> relatedProject = resource.getRelatedProject();
+            if (relatedProject.isPresent()) {
+                event.getPresentation().setVisible(true);
```

Comment:
Earlier `setVisible(true)` was invoked each time. In your case only if `relatedProject.isPresent()`. Are you sure that you should rework it?